packages/compass-sidebar/src/components/non-genuine-warning-modal/non-genuine-warning-modal.jsx

```diff
@@ -1,91 +1,92 @@
-export const MODAL_TITLE = 'Non-Genuine MongoDB Detected';
+const bannerStyles = css({
+  marginTop: spacing[4]
+});
 
-/**
- * Component for the non-genuine MongoDB warning modal.
- */
-class NonGenuineWarningModal extends PureComponent {
-  static displayName = 'NonGenuineWarningModal';
-  static propTypes = {
-    isVisible: PropTypes.bool.isRequired,
-    toggleIsVisible: PropTypes.func.isRequired,
-    openLink: PropTypes.func.isRequired
-  };
+const modalContentStyles = css({
+  padding: spacing[5]
+});
 
-  /**
-   * Close modal.
-   *
-   * @param {Object} evt - The click event.
-   */
-  handleClose(evt) {
-    if (evt) {
-      evt.preventDefault();
-      evt.stopPropagation();
-    }
-    this.props.toggleIsVisible(false);
-  }
+const modalBodyStyles = css({
+  marginTop: spacing[3],
+  marginBottom: spacing[2]
+});
 
-  /**
-   * Render the non-genuine mongodb warning modal.
-   *
-   * @returns {React.Component} The non-genuine warning modal.
-   */
-  render() {
-    return (
-      <Modal
-        // Because this modal is rendered outside of the
-        // react root we need to apply the deprecated bootstrap styles here.
-        className="with-global-bootstrap-styles"
-        show={this.props.isVisible}
-        backdrop="static"
-        dialogClassName={styles['non-genuine-warning-modal']}
-        onHide={this.handleClose.bind(this)}
-      >
-        <Modal.Header>
-          <Modal.Title>{MODAL_TITLE}</Modal.Title>
-        </Modal.Header>
+const DESCRIPTION = 'Some documented MongoDB features may work differently, be entirely missing'
+  + ' or incomplete, or have unexpected performance characteristics. ';
+const WARNING_BANNER = 'This server or service appears to be an emulation of MongoDB rather than an official MongoDB product.';
+const LEARN_MORE_URL = 'https://docs.mongodb.com/compass/master/faq/#how-does-compass-determine-a-connection-is-not-genuine';
+const MODAL_TITLE = 'Non-Genuine MongoDB Detected';
 
-        <Modal.Body>
-          <div className={styles['non-genuine-warning-modal-message']}>
-            <FontAwesome name="exclamation-circle"/>
-            &nbsp; {WARNING_BANNER} &nbsp;
-          </div>
+function NonGenuineWarningModal({
+  isVisible,
+  toggleIsVisible,
 
-          <div className={styles['non-genuine-warning-modal-p1']}>
-            {P1}
-            <a
-              onClick={() => this.props.openLink(LEARN_MORE_URL)}
-              data-test-id="non-genuine-warning-modal-learn-more-link">
-              Learn more
-            </a>&nbsp;
-          </div>
-        </Modal.Body>
+}) {
+  const onClose = useCallback(() => {
+    toggleIsVisible(false);
+  }, [toggleIsVisible]);
 
-        <Modal.Footer>
-          <TextButton
-            className="btn btn-primary btn-sm"
-            dataTestId="continue-button"
-            text="CONTINUE"
-            clickHandler={this.handleClose.bind(this)} />
-        </Modal.Footer>
-      </Modal>
-    );
-  }
+  return (
+    <Modal
```

Comment:
Isn't the confirmation modal appropriate for this use case? https://www.mongodb.design/component/confirmation-modal/example/

Reply:
I was thinking Confirmation Modal fits most of the use case, however, in the previous designs we only have one action - `Continue` - if we would use the confirmation modal it would add a `Cancel` button which we can't conditionally show. We can use it if we're alright moving away from the current single action flow we have. Do you have any preference? I can ping Claudia.

Reply:
I see, seems ok to don't use `ConfirmationModal` if it doesn't do what we want.